### PR TITLE
QuickAccess: add quick note

### PR DESCRIPTION
**Pull request by feloxx**
Impl: #2351 

In this modification, the function of double-click tab to quickly create notes is added. do not adjust the template for now.

### CODE REVIEW

File: src/widgets/viewsplit.cpp [hunk 1]
```diff
@@ -711,6 +715,16 @@ void ViewSplit::createContextMenuOnTabBar(QMenu *p_menu, int p_tabIdx)
     }
 }
 
+void ViewSplit::newTab(int p_idx)
```
Comment (tamlok) [inline src/widgets/viewsplit.cpp:718]: The name here is confusing. Rename it to `CreateQuickNote` without the `p_idx`.

File: src/core/notebook/node.h [hunk 2]
```diff
@@ -207,6 +209,8 @@ namespace vnotex
         Node *m_parent = nullptr;
 
         QVector<QSharedPointer<Node>> m_children;
+
+        QString m_manualAbsolutePath;
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/core/notebook/node.h:213]: Deleted.

File: src/widgets/notebookexplorer.cpp [hunk 3]
```diff
@@ -264,23 +267,80 @@ void NotebookExplorer::newFolder()
     }
 }
 
-void NotebookExplorer::newNote()
+void NotebookExplorer::newNote(const QVector<int> &p_type, const QString &p_path)
 {
-    auto node = checkNotebookAndGetCurrentExploredFolderNode();
-    if (!node) {
-        return;
+    if (p_type.length() <= c_defaultCreateNote) {
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/notebookexplorer.cpp:272]: It is mainly used to determine p_type, Is it the default, is it single, is it multiple.

File: src/widgets/notebookexplorer.cpp [hunk 4]
```diff
@@ -264,23 +267,80 @@ void NotebookExplorer::newFolder()
...
+        if (dialog.exec() == QDialog::Accepted) {
+            m_nodeExplorer->setCurrentNode(dialog.getNewNode().data());
+
+            // Open it right now.
+            auto paras = QSharedPointer<FileOpenParameters>::create();
+            paras->m_mode = ViewWindowMode::Edit;
+            paras->m_newFile = true;
+            emit VNoteX::getInst().openNodeRequested(dialog.getNewNode().data(), paras);
+        }
+    } else if (p_type.length() == c_singleQuickNote)  {
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/notebookexplorer.cpp:287]: Want to use  `< 1` ，`= 1`, but that's a magic constant, which is hard to understand

File: src/widgets/viewsplit.cpp [hunk 5]
```diff
@@ -80,7 +81,10 @@ void ViewSplit::setupUI()
                 closeTab(p_idx);
             });
     connect(this, &QTabWidget::tabBarDoubleClicked,
-            this, &ViewSplit::closeTab);
+            this, [this](int p_idx) {
+                newTab(p_idx);
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/viewsplit.cpp:85]: Done.

File: src/widgets/viewsplit.cpp (hunk 1 above)
Comment (feloxx) [reply src/widgets/viewsplit.cpp:718]: Done.

File: src/widgets/viewsplit.cpp [hunk 6]
```diff
@@ -711,6 +715,16 @@ void ViewSplit::createContextMenuOnTabBar(QMenu *p_menu, int p_tabIdx)
     }
 }
 
+void ViewSplit::newTab(int p_idx)
+{
+    if (p_idx == -1) {
+        auto &sessionConfig = ConfigMgr::getInst().getSessionConfig();
```
[the remark this answers is not in the document]
Comment (feloxx) [reply src/widgets/viewsplit.cpp:721]: Done.

**Pull request merged**